### PR TITLE
refactored-code

### PR DESCRIPTION
**Pull request by mudasiri**
In this pull request I implemented the following:

- Refactored my Component class into a function
- Added hooks and state

### CODE REVIEW

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @mudasiri, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.

## 🛑Required Changes ♻️
_Check the comments under the review._

- [ ] The design does not meet the requirements, It was difficult for me to see some answers, I know this was not a requirement for this milestone, but kindly work on the styling.

styling


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- In order to make the calculator component available for review, make few changes to that function. For example, you can use ES6 syntax like const for method decralation.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mudasiri** [comment]
I have updated my codes and added borders around.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
## Hi @mudasiri, 

Great work on making the changes requested by a previous reviewer 👏 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
stay-strong-youre-almost-there


 # Status: Requires Changes ♻️
_Check the comments below and under the review._

Please do consider, as requested by the previews reviewer:
- [ ] Fix the display of your calculator, checkout the screenshot below and kindly fix the issue. 

display_error

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.css [hunk 1]
```diff
@@ -108,6 +110,21 @@ button {
...
+.normal {
+  border-right: 1px solid white !important;
+  border-bottom: 1px solid white !important;
+}
+
+.special {
+  border-bottom: 1px solid white !important;
+  border-right: 0 solid white !important;
+}
+
```
Comment (zieeco) [inline src/App.css:127]: #### REFACTOR USING HOOKS (APP.JS)

- Hey, you're yet to refactor the app.js, please, kindly fix this issue below:


Screenshot from 2023-01-20 13-56-29

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @mudasiri,


Great job so far!  👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.css (hunk 1 above)
Comment (mudasiri) [reply src/App.css:127]: @zieeco  The Project requirements didn't mention refactor of App.js. Please see below the project requirement that specifically mentions only calculator.js .

Project requirements

Refactor the Calculator component (components/Calculator.js) from a class based component to a functional component with hooks.
After the refactor is done, test the app in the browser. It should keep the same functionality.

I think your review comment should be optional and not compulsory before passing. Kindly reconsider your decision.

Comment (zieeco) [reply src/App.css:127]: Hey @mudasiri, Please, kindly refactor the app.js, though it wasn't mentioned, you're meant to refactor it as well. 
Sorry for the inconvenience.

Comment (mudasiri) [reply src/App.css:127]: @zieeco  I have refactored the app.js as you have recommended. Kindly check and approve. Thanks

**Review by rotimiazeez — APPROVED** [review]
Hi @mudasiri 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good communication with CRs 👏🏽
- No Linter error ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. The previous reviewer missed some of them, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something needs to be 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**